File: client/acmq.py
import json
from socket import socket, AF_INET, SOCK_STREAM


class ACMQ():
	def __init__(host,port):
		self.host = host
		self.port = port

	def push(body):
		string = json.dumps(body)
		string = "PUSH %s" % string
		return_value = transact(string)
		if return_value != "ACK\n":
			raise "Pushing failed %s" % return_value
		return return_value

	def pop():
		string = transact("POP\n")
		return json.loads(string)[0]

	def transact(string):
		s = socket(AF_INET, SOCK_STREAM)
		s.connect((self.host, self.port))
		s.send(string)
		data = s.recv(1024)

File: client/acmq.rb
require 'json'
require 'socket'

class ACMQ
  def initialize(host, port)
    @host = host
    @port = port
  end

  def push (body)
    str = JSON.dump [body]
    str = "PUSH " + str
    return_value = transact(str)
    if return_value != "ACK\n"
      raise "Pushing failed" + return_value
    end
  end

  def pop
    str = transact("POP\n")
    JSON.parse(str)[0]
  end

  def transact (str)
    s = TCPSocket.new @host, @port
    s.puts(str)
    s.gets
  end
end
